jerry-core/parser/js/js-scanner-util.c

```diff
@@ -564,14 +564,23 @@ scanner_pop_literal_pool (parser_context_t *context_p, /**< context */
       {
         no_declarations++;
 
-#if !ENABLED (JERRY_ES2015)
+#if ENABLED (JERRY_ES2015)
+        if ((type & (SCANNER_LITERAL_IS_CONST | SCANNER_LITERAL_IS_ARG)) == SCANNER_LITERAL_IS_CONST)
```

Comment:
If a reviewer insists, the two code paths can be merged (using a `condition` boolean), especially because the assert is true in the other case as well. It feels ugly for me, but I will not object.